geoh5py/objects/text.py

```diff
@@ -0,0 +1,192 @@
+#                                                                              '
+#  You should have received a copy of the GNU Lesser General Public License    '
+#  along with geoh5py.  If not, see <https://www.gnu.org/licenses/>.           '
+# ''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''''
+
+
+from __future__ import annotations
+
+import uuid
+from json import loads
+from typing import TYPE_CHECKING
+
+import numpy as np
+from pydantic import (
+    AliasChoices,
+    AliasGenerator,
+    BaseModel,
+    ConfigDict,
+    Field,
+    field_validator,
+)
+from pydantic.alias_generators import to_pascal, to_snake
+
+from .points import Points
+
+
+if TYPE_CHECKING:
+    from numpy import ndarray
+
+
+class TextEntry(BaseModel):
+    """
+    Core parameters for text mesh data.
+    """
+
+    model_config = ConfigDict(
+        alias_generator=AliasGenerator(  # type: ignore
+            validation_alias=to_snake,
+            serialization_alias=to_pascal,
+        )
+    )
+    starting_point: str
+    annotation: int = 0
+    color: str = "#ffffff"
+    direction: str = "{1,0,0}"
+    font: str = "builtin_simple"
+    font_size: int = 10
+    layer_name: str = ""
+    normal: str = "{0,0,1}"
+    obliquing_angle: int = 0
+    rotation_angle: int = 0
+    text: str = "<no-data>"
+    width_scale_factor: float = 1.0
+
+    @field_validator("starting_point", "normal", "direction", mode="before")
+    @classmethod
+    def array_to_str(cls, float_list: np.ndarray | list):
+        if isinstance(float_list, np.ndarray | list):
+            float_list = "{" + ",".join([str(val) for val in float_list]) + "}"
+
+        return float_list
+
+
+class TextData(BaseModel):
+    """
+    Text mesh data.
+    """
+
+    text_data: list[TextEntry] = Field(
+        validation_alias=AliasChoices("text_data", "Text Data"),
+        serialization_alias="Text Data",
+    )
+
+
+class TextObject(Points):
+    """
+    Text object for annotated text labels in viewport.
+    """
+
+    _attribute_map: dict = Points._attribute_map.copy()
+    _attribute_map.update({"TextMesh Data": "text_mesh_data"})
+    _default_name = "Text Object"
+    _TYPE_UID: uuid.UUID = uuid.UUID("{c00905d1-bc3b-4d12-9f93-07fcf1450270}")
+
+    def __init__(
+        self,
+        vertices: np.ndarray | list | tuple | None = None,
+        text_mesh_data: str | None = None,
+        **kwargs,
+    ):
+        self._text_mesh_data: TextData
+
+        super().__init__(vertices=vertices, text_mesh_data=text_mesh_data, **kwargs)
+
+    @property
+    def text_mesh_data(self) -> TextData:
+        """
+        Text mesh data.
+        """
+
+        return self._text_mesh_data
+
+    @text_mesh_data.setter
+    def text_mesh_data(self, value: str | None):
+        """
+        Set the text mesh data.
+        """
+
+        self._text_mesh_data = self._validate_text_data(value)
+
+    @property
```

Comment:
dfo some trivial test for extent is None (vallable for all None returning functions)